app/models/referee_interface/reference_feedback_form.rb

```diff
@@ -20,12 +20,21 @@ def save
         # state of the application choices
         if reference.application_form.application_references_complete?
           reference.application_form.application_choices.includes(:course_option).find_each do |application_choice|
-            ApplicationStateChange.new(application_choice).references_complete!
+            complete_references(application_choice)
           end
         end
       end
 
       true
     end
+
+  private
+
+    def complete_references(application_choice)
+      ApplicationStateChange.new(application_choice).references_complete!
+      if application_choice.edit_by <= Time.zone.now
+        SendApplicationToProvider.new(application_choice: application_choice).call
+      end
```

Comment:
Would it be possible to schedule a `SendApplicationsToProviderWorker` instead, so that we don't duplicate the logic?

Reply:
Yes we could do that - it will make the process slightly less immediate from the providers point of view but normally the workers should be pretty quick so I guess it's fine.

Reply:
On second thoughts I think it should stay as is.

If we schedule a worker I'm not sure what we would do with the various specs etc. that invoke `ReceiveReference` to move the status on. We couldn't just drop `ReferenceFeedbackForm` in as a replacement. And we could end up with multiple instances of that worker running in parallel which is likely to cause errors (unless we specifically guard against that somehow).